### ISSUE TITLE
[Project Wizard] Template loading should use wizard progress bar

### ISSUE THREAD

**Issue by lak-proddev** [eclipse/dartboard#118]
After selecting the "Use Stagehand temple" option, the wizard page is showing the custom progress bar. I think we should use the wizard progress bar instead of a custom bar.

Generally, no eclipse wizards show custom progress bars.
![image](https://user-images.githubusercontent.com/39690425/62511716-9178cf00-b832-11e9-900f-69fb22ab0370.png)

**Comment by jonas-jonas** [comment]
If I see correctly the built-in progress bar only allows to add more progress to it? I like that this one just goes from side to side.

Also It's nice to have it be explicitly inside the Stagehand group to signal that there is a task that only affects that functionality, not the whole dialog.

Feel free to submit a PR for this, as I don't know how to implement this right now.

**Comment by lak-proddev** [comment]
Jonas,  I modified the code to use progress monitor from wizard instead of custom progress.
I did some basic testing. Please take care if anything missing. 
You can improve the monitor by calculating and modify `subMonitor.worked(..);` values in StagehandService class.

![image](https://user-images.githubusercontent.com/39690425/62544762-0a514880-b87e-11e9-86de-db204a0cd359.png)

**Comment by jonas-jonas** [comment]
> You can improve the monitor by calculating and modify subMonitor.worked(..); values in StagehandService class.

That's the issue though. We don't know these values upfront, and I hate these progressbars that jump from 0-50% in seconds and than take ages to complete. 

And as I said, with your screenshot it looks like there is something to do before the whole wizard is _usable_ but that's not true, you can create projects without stagehand and the gathering job doesn't need to complete before you _can_ click _Finish_.

I don't know, the progress bar was intended to give the user a hint about why the __Use Stagehand template__ in particular is not clickable. I'd like to keep it that way, even if that means "defying" the usual "Eclipse-way-of-things" (which is far from being always right!). @vogella WDYT about this?

**Comment by lak-proddev** [comment]
Frankly speaking, we really don't need progress bar here. The progress monitor intended for long-running operations. 
Here the long-running operation is `StagehandService #`getStagehandTemplates()`  that also only the first time when it is called. Later it will return cache values.

I think we can put loading operation in the `init` method of the wizard as a separate thread.

**Comment by jonas-jonas** [comment]
Okay, but will it show an indicator? That's why it is there in the first place. The point is not to show __the__ actual progress, but rather that _some_ progress is currently being made.

See how the intellij plugin does it:
![ij-stagehand](https://user-images.githubusercontent.com/5540255/62548492-2f868c80-b867-11e9-8d7a-56b3baf91998.gif)

**Comment by lak-proddev** [comment]
Used unknown flag to show continuous progress. Since difficult to determine the size of units.
https://gyazo.com/ce808d6168296ce3447ca88ed7c8face

**Comment by jonas-jonas** [comment]
Hm, thanks for the gif. I still don't like the look of it since you can't click next when the indicator is still loading. Which shouldn't be the case, as you don't have to use the combo at all when using the wizard. Is there any downside to using a custom indicator?

**Comment by jonas-jonas** [comment]
The current "custom" progress should be kept unless there is some usability or performance implication in doing so. Please let me know then.
